### DISCUSSION TITLE
`rep(1)` not working (empty)

### DISCUSSION THREAD

**Discussion by axieax** [L3MON4D3/LuaSnip#533] [Q&A] [answered]
Hi, I'm trying to make a snippet for a Lua module. However, I'm not sure how to repeat the highlighted node with index 1 from the previous / upper level (not in the same curly brace). Is there a way I can achieve this?

![image](https://user-images.githubusercontent.com/62098008/185098387-613005d5-e282-4b9a-a7e2-8878cd0605fd.png)

**Comment by L3MON4D3** [comment, marked as answer]
Uhh nice, a usecase for `absolute_indexer` :D
Passing (with `local ai = require("luasnip.nodes.absolute_indexer")`) `ai[1]` instead of `1` should do the trick.
Check in DOC.md or `:h luasnip-absolute_indexer` for more info

**Reply by axieax** [reply]
Thank you so much!! I've updated my code to the following, but for some reason, jumping to the next position is skipping positions (c, 2). Do you have any ideas on this?

**Reply by L3MON4D3** [reply]
Mhmmm, could you show a gif? Can't quite picture what's happening :sweat_smile:

**Reply by axieax** [reply]
Yeah kinda hard to explain, apologies for the confusions!

https://user-images.githubusercontent.com/62098008/185149312-275b888c-17e0-4ddb-93e7-1dbb02f0b962.mp4

**Reply by L3MON4D3** [reply]
Ahhhh alright: the choiceNode is jumped into, but there is no position at which the cursor could stop (ie. an insertNode) inside it, so it's just jumped over.
Adding an `i(1)` somewhere (the beginning, probably) should prevent that

**Reply by axieax** [reply]
Do you mean something like this? Not sure why it's still not working rip

```lua
local mod = s(
  "mod",
  fmt(
    [[
    local {} = {{}}
    {}
    return {}
    ]],
    {
      i(1, "M"),
      c(2, {
        i(1),
        t(""),
        fmt(
          [[
            function {}.config()
            end
          ]],
          { rep(ai[1]) }
        ),
        fmt(
          [[
            function {}.setup()
            end

            function {}.config()
            end
          ]],
          { rep(ai[1]), rep(ai[1]) }
        ),
      }),
      c(3, {
        i(1),
        rep(ai[1]),
        fmt("setmetatable({}, {{}})", { rep(ai[1]) }),
      }),
    }
  )
)
```

**Reply by L3MON4D3** [reply]
Ahh, "no position to stop in the active choice" is what I should've said.
so
```lua
fmt(
          [[
            function {}.config()
            end
          ]],
          { rep(ai[1]) }
```
to
```lua
fmt(
          [[
            {}function {}.config()
            end
          ]],
          { i(1), rep(ai[1]) }
```

**Reply by axieax** [reply]
Ahh I got it now! Cheers for the followup :))
